### ISSUE TITLE
Possible incorrect logic with registering new Player

### ISSUE THREAD

**Issue by victordidenko** [videojs/video.js#8925]
I'm not sure, but shouldn't there be `some` instead of `every`?

"If at least one player is not disposed yet"

https://github.com/videojs/video.js/blob/62f38446a5f01e85739bdb7207c7e992b57c1c86/src/js/component.js#L2024-L2032

**Comment by gkatsev** [comment]
Yes, I think you're correct that that should be a some and not every. Care to submit a PR? (Ideally, with a new test for it. Can use this as a template https://github.com/videojs/video.js/blob/main/test/unit/player.test.js#L2328-L2346)

Thanks!

**Comment by victordidenko** [comment]
Sure, but I wanted to ask about what is the reason of this limitation?
Because, funny enough, I found it while trying to _workaround_ this error.

**Comment by gkatsev** [comment]
It's to try and make sure that all player instances on the page use the same player component. The idea being that if there are players that aren't the same player component, there may end up issues around it.

**Comment by victordidenko** [comment]
Yeah, I understand _what_ this check is doing, I don't understand _why_ :)
This exactly — "there may end up issues around it" — what issues?
My only guess is that this is because of `Player.players` map? It might be used somewhere else, maybe.
But if new Player class is a subclass of original Player class — they shares the same `.players` map, so, it shouldn't be a problem? 🤔
Maybe this check should be more comprehensive than current version?

**Comment by gkatsev** [comment]
Yes, this error is a bit theoretical. But, we essentially never want to get an issue opened because a user initialized a player, set a new player component, and then initialized a new player. Being able to override the player component is itself a pretty new capability. It used to be explicitly disallowed, I believe.

I think the thought is that if someone is iterating through all the players, which hooks in video.js do, it's possible that different player components may or may not have the same functionality. So, you may break if you're expecting a method, but it isn't available on all instances.

**Comment by gkatsev** [comment]
Can I ask what you're using the custom player for? And what issue did you encounter that you hit this block? Maybe there are alternatives.

**Comment by victordidenko** [comment]
@gkatsev Made a PR #8932
